### PR TITLE
Make it possible to clone and merge wxGridCellAttr

### PR DESCRIPTION
**Pull request by frodegill**
Hope this should cover most of the issue...

### CODE REVIEW

File: interface/wx/grid.h [hunk 1]
```diff
@@ -22,7 +22,7 @@
          wxGridCellFloatRenderer, wxGridCellNumberRenderer,
          wxGridCellStringRenderer
 */
-class wxGridCellRenderer : public wxClientDataContainer, public wxRefCounter
+class wxGridCellRenderer : public wxSharedClientDataContainer, public wxRefCounter
```
Comment (vadz) [inline interface/wx/grid.h]: We need to document `wxSharedClientDataContainer` too if it appears in the documentation, it should be self-consistent.

File: src/ribbon/gallery.cpp [hunk 2]
```diff
@@ -36,7 +36,7 @@ wxDEFINE_EVENT(wxEVT_RIBBONGALLERY_CLICKED, wxRibbonGalleryEvent);
 wxIMPLEMENT_DYNAMIC_CLASS(wxRibbonGalleryEvent, wxCommandEvent);
 wxIMPLEMENT_CLASS(wxRibbonGallery, wxRibbonControl);
 
-class wxRibbonGalleryItem
+class wxRibbonGalleryItem : public wxSharedClientDataContainer
```
Comment (vadz) [inline src/ribbon/gallery.cpp]: Not sure if this is really needed. I guess I don't have any real objections to it neither, but as long as the items are not copied/cloned, this just doesn't seem necessary. But we can keep it if you see value in it.

**Comment by vadz** [comment]
Thanks for the updates, this looks almost perfect and we just need to document the new class before this can be merged.

**Comment by frodegill** [comment]
I've added documentation. Is there anything more I should do?

**Comment by vadz** [comment]
Sorry for the lack of review for your last commit!

Now that I look at it, I'm not sure if I'm missing something or if there has been a misunderstanding: by documentation, I mean Doxygen comments in `interface/wx/grid.h`, this is the source of the official documentation. Comments in the real header are nice to have, of course, but not sufficient.

As I mentioned in the comment for `wxGridCellRenderer`, the new class needs to be present in the documentation header as it appears as a base class there -- so it also needs to be at least minimally defined, otherwise various tools using these headers as their input would choke on it.

Please let me know if anything is still unclear and TIA!

**Comment by vadz** [comment]
I'm sorry but it looks like this is never going to be applied, after languishing here for several years already. I don't really have the motivation to document it myself and, frankly, I'm not sure why do we want to do all this at all (I may be forgetting something, but has this ever been discussed/explained?).

**Pull request closed without merging**